gc/base/MemoryManager.cpp

```diff
@@ -487,6 +500,23 @@ MM_MemoryManager::createVirtualMemoryForMetadata(MM_EnvironmentBase* env, MM_Mem
 	return NULL != handle->getVirtualMemory();
 }
 
+void
+MM_MemoryManager::destroyVirtualMemoryForHeap(MM_EnvironmentBase* env, MM_MemoryHandle* handle)
+{
+	destroyVirtualMemory(env, handle);
+
+#if defined(OMR_ENV_DATA64) && !defined(OMR_GC_COMPRESSED_POINTERS)
+	MM_GCExtensionsBase* extensions = env->getExtensions();
+	MM_VirtualMemory* shadowMemory = extensions->shadowHeapHandle.getVirtualMemory();
+	if (NULL != shadowMemory) {
+
+		shadowMemory->kill(env);
+		extensions->shadowHeapHandle.setVirtualMemory(NULL);
+	}
```

Comment:
should we also set `extensions->shadowHeapBase` and `extensions->shadowHeapTop` to NULL here?